crates/optimism/primitives/src/transaction/signed.rs

```diff
@@ -0,0 +1,440 @@
+//! A signed Optimism transaction.
+
+use core::mem;
+
+use alloy_consensus::{
+    transaction::RlpEcdsaTx, SignableTransaction, TxEip1559, TxEip2930, TxEip7702,
+};
+use alloy_eips::{
+    eip2718::{Decodable2718, Eip2718Error, Eip2718Result, Encodable2718},
+    eip2930::AccessList,
+    eip7702::SignedAuthorization,
+};
+use alloy_primitives::{
+    keccak256, Address, Bytes, PrimitiveSignature as Signature, TxHash, TxKind, Uint, B256,
+};
+use alloy_rlp::Header;
+use derive_more::{AsRef, Constructor, Deref};
+use op_alloy_consensus::{OpTypedTransaction, TxDeposit};
+#[cfg(any(test, feature = "reth-codec"))]
+use proptest as _;
+use reth_primitives::{
+    transaction::{recover_signer, recover_signer_unchecked},
+    TransactionSigned,
+};
+use reth_primitives_traits::{InMemorySize, SignedTransaction};
+use serde::{Deserialize, Serialize};
+
```

Comment:
```suggestion
```

Reply:
this keeps local deps separate from deps from other crates so that cohesion in the crate is clear to the human eye